File: polimorfismo/hr-csharp/Course.OOP.Polimorfism.HR/Entities/Manager.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Course.OOP.Polimorfism.HR.Entities
{
    public class Manager : Employee
    {

        public double Bonus { get; set; }


        /*
         * Si se desea que el método utilizado sea el de la clase padre
         * cuando el tipo de variable sea Employee, entonces cambiar
         * el keyword de override a new.
         */
        public override double CalculateNetSalary()
        {

            Console.WriteLine("Estamos en la clase Manager");

            double baseSalary = base.CalculateNetSalary();
            baseSalary += Bonus;
            return baseSalary;
        }

    }
}

File: polimorfismo/hr-java/src/main/java/course/oop/polimorfism/entities/Manager.java
package course.oop.polimorfism.entities;

import java.time.LocalDate;

public class Manager extends Employee {

    private double bonus;

    public Manager(int employeeID, String firstName, String lastName, LocalDate startDate, double baseSalary, double bonus) {
        super(employeeID, firstName, lastName, startDate, baseSalary);
        this.bonus = bonus;
    }

    public double calculateNetSalary()
    {
        System.out.println("Estamos en la clase Manager");
        double baseSalary = super.calculateNetSalary();
        baseSalary += bonus;
        return baseSalary;
    }
}
